### PR TITLE
feat(geo): save search results in layer

### PR DESCRIPTION
**Pull request by aziz-access**
**What is the current behavior?** (You can also link to an open issue here)
This update relate to the issue [896](https://github.com/infra-geo-ouverte/igo2/issues/896) in igo2.

**What is the new behavior?**
Save search results to an existing or a new layer.

### CODE REVIEW

File: packages/geo/src/lib/search/search-results/search-results-add-button.component.ts [hunk 1]
```diff
@@ -57,7 +84,17 @@ export class SearchResultAddButtonComponent implements OnInit, OnDestroy {
   }
   private _color = 'primary';
 
-  constructor(private layerService: LayerService) {}
+  @Input() stores: FeatureStore<FeatureWithDraw>[] = [];
+
+  get allLayers() {
+    return this.map.layers.filter((layer) =>
+      layer.id.includes('igo-saved-layer')
```
Comment (PhilippeLafreniere18) [inline packages/geo/src/lib/search/search-results/search-results-add-button.component.ts:91]: I think "igo-search-layer" would be more meaningful

File: packages/geo/src/locale/en.geo.json [hunk 2]
```diff
@@ -12,6 +12,7 @@
         "unavailableTitle": "Catalog unavailable",
         "unavailable": "The catalog '{{value}}' is unavailable at the moment or you don't have the required permissions.",
         "someUnavailable": "Some of the layers from the current catalog are unavailable at the moment or you don't have the required permissions.",
+        "feature.addToLayar": "Add to Layar",
```
Comment (PhilippeLafreniere18) [inline packages/geo/src/locale/en.geo.json:15]: Same as above (change to search section, remove 'feature.')

**Review by PhilippeLafreniere18 — CHANGES_REQUESTED** [review]
Without testing, just a few minor comments

File: packages/geo/src/lib/search/search-results/search-results-add-button.component.ts [hunk 3]
```diff
@@ -12,18 +12,43 @@ import { LayerOptions } from '../../layer/shared/layers/layer.interface';
...
   changeDetection: ChangeDetectionStrategy.OnPush
 })
-export class SearchResultAddButtonComponent implements OnInit, OnDestroy {
+export class SearchResultAddButtonComponent implements OnInit, OnDestroy{
   public tooltip$: BehaviorSubject<string> = new BehaviorSubject(
     'igo.geo.catalog.layer.addToMap'
   );
 
+  public exportTooltip$: BehaviorSubject<string> = new BehaviorSubject(
+    'igo.geo.catalog.feature.addToLayar'
```
[the remark this answers is not in the document]
Comment (aziz-access) [reply packages/geo/src/lib/search/search-results/search-results-add-button.component.ts:49]: ok

File: packages/geo/src/lib/search/search-results/save-feature-dialog.component.html [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+        color="accent"
+        [result]="feature"
+        showIcons="true">
+      </igo-search-results-item>
+    </igo-list>
+    
+    <div class="igo-input-container">
+      <mat-form-field>
+        <input type="text"
+          placeholder="{{'igo.geo.layer.pleaseChooseOrSet' | translate}}"
```
[the remark this answers is not in the document]
Comment (aziz-access) [reply packages/geo/src/lib/search/search-results/save-feature-dialog.component.html:17]: ok

File: packages/geo/src/lib/search/search-results/search-results-add-button.component.ts [hunk 5]
```diff
@@ -12,18 +12,43 @@ import { LayerOptions } from '../../layer/shared/layers/layer.interface';
...
   changeDetection: ChangeDetectionStrategy.OnPush
 })
-export class SearchResultAddButtonComponent implements OnInit, OnDestroy {
+export class SearchResultAddButtonComponent implements OnInit, OnDestroy{
   public tooltip$: BehaviorSubject<string> = new BehaviorSubject(
     'igo.geo.catalog.layer.addToMap'
   );
 
+  public exportTooltip$: BehaviorSubject<string> = new BehaviorSubject(
+    'igo.geo.search.addToLayar'
```
Comment (PhilippeLafreniere18) [inline packages/geo/src/lib/search/search-results/search-results-add-button.component.ts:49]: typo -> addToLayer

File: packages/geo/src/locale/en.geo.json [hunk 6]
```diff
@@ -424,6 +428,7 @@
         "layer.placeholder": "Search for a layer",
         "ichercheReverse.name": "Search by coordinates",
         "clearSearch": "Clear search",
+        "addToLayar": "Add to Layar",
```
Comment (PhilippeLafreniere18) [inline packages/geo/src/locale/en.geo.json:431]: typo -> addToLayer

File: packages/geo/src/locale/fr.geo.json [hunk 7]
```diff
@@ -423,6 +427,7 @@
         "layer.placeholder": "Rechercher une couche de données.",
         "ichercheReverse.name": "Recherche par coordonnées",
         "clearSearch": "Effacer la recherche",
+        "addToLayar": "Ajouter à une couche",
```
Comment (PhilippeLafreniere18) [inline packages/geo/src/locale/fr.geo.json:430]: typo -> addToLayer

**Review by PhilippeLafreniere18 — APPROVED** [review]

**Review by PhilippeLafreniere18 — CHANGES_REQUESTED** [review]
Two problems here : 
- I hit the following error when adding a new search feature to an existing search layer
image
 - It seems to have a problem with z index criteria. Layer list should be sorted by z index, but when adding search layer, weird behavior is happening
 
image
image

**Review by PhilippeLafreniere18 — APPROVED** [review]

**Pull request merged**